lab8/8-4.js

```diff
@@ -0,0 +1,50 @@
+(function() {    
+    var input = prompt('input your card number'),
+        card = parseInt(input),
+        nDigits = Math.log10(card) + 1 | 0, //valid for positive integers
+        cardType = 'INVALID';
+   
+    if (nDigits == 15 && (card >= 340000000000000 || card >= 370000000000000)) {
+        cardType = 'AMEX';
+    }
+    else if (nDigits == 16 && (card >= 5100000000000000 || card >= 5200000000000000 || card >= 5300000000000000 
+            || card >= 5400000000000000 || card >= 5500000000000000)) {
```

Comment:
Again, the first check covers everything: from 5100000000000000 to 9999999999999999.

Reply:
ok, you're right. 
fixed: (card >= 4000000000000 && card <= 4999999999999)